### ISSUE TITLE
No instructions for Demo

### ISSUE THREAD

**Issue by wkiess01** [MicrosoftDocs/xamarin-docs#2229]
I've downloaded the demo files and opened the solution in VS. However, I can't debug the solution as there is no main app to run and VS complains. It would have been nice to have full instructions on how to install and run the demo files so newbies like myself can actually get the code running and see what it does an how. You can't take it for granted everyone knows how to run the demos.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: e94288f0-f661-6a8c-bf43-01332272a777
* Version Independent ID: 5bd102bf-ffee-c2f9-4abc-27f14111bb21
* Content: [SkiaSharp Graphics in Xamarin.Forms - Xamarin](https://docs.microsoft.com/en-us/xamarin/xamarin-forms/user-interface/graphics/skiasharp/#feedback)
* Content Source: [docs/xamarin-forms/user-interface/graphics/skiasharp/index.md](https://github.com/MicrosoftDocs/xamarin-docs/blob/live/docs/xamarin-forms/user-interface/graphics/skiasharp/index.md)
* Product: **xamarin**
* Technology: **xamarin-skiasharp**
* GitHub Login: @davidbritch
* Microsoft Alias: **dabritch**

**Comment by davidbritch** [comment]
Hi @wkiess01 

The sample (https://docs.microsoft.com/samples/xamarin/xamarin-forms-samples/skiasharpforms-demos/) consists of four projects:

- A .NET Standard library.
- An iOS project.
- An Android project.
- A UWP project.

The iOS/Android/UWP projects are all runnable, provided you have those platforms installed within Visual Studio. For more info, see [Installing Xamarin](https://docs.microsoft.com/xamarin/get-started/installation/?pivots=windows).

**Comment by wkiess01** [comment]
Hi David,

Thank you for your reply. Please forgive me but I am truly a newbie at
this, C#, Android, Xamarin.Since VS was complaining about a startup project
I set it to the Android Project. When I tried to debug it, it built the
project but then there was the following error message in the error list:

Severity Code Description Project File Line Suppression State
Error MSB3231: Unable to remove directory
"obj\Debug\90\android\bin\classes". Could not find a part of the path
'AccessibilityManagerCompat_AccessibilityStateChangeListenerImplementor.class'.
0

I simply have no idea what that means or how to rectify it.

The steps I took to get this:
1) Download the demo zip file and extract it to a folder in my Xamarin
tutorials folder
2) Started VS 2019 Community Edition
3) "Open a Project or Solution" and navigate to the demo folder's solution
4) Set the startup project as the Android project
5) Started the Android Emulator
6) Press F5 to start debugging the Android project
7) Received a message advising "There were deployment errors"
8) The above message is shown in the error list.

I can't remember if I had to install  SkiaSharp, SkiaSharp.Views &
SkiaSharp.Views.Forms before or after trying this once, but they're
definitely installed, V1.60.0 according to the nuget package manager (I
don't know what his is either).

Cheers,
Walter

**Comment by davidbritch** [comment]
Hi @wkiess01 

It sounds like a problem with your Xamarin install within Visual Studio. I recommend you try [this](https://docs.microsoft.com/xamarin/get-started/first-app/?pivots=windows) tutorial to verify whether you can deploy a simple application to the Android emulator. If you can, great. If you can't, you have a problem with your Xamarin install.

**Comment by wkiess01** [comment]
Thanks David.

It turned out to be the length of the pathname for the demo project files
was too long. It seems Windows 10 still has a pathetic low pathname length
before it starts spitting the dummy. I moved the demo project out of the
tutorial folder structure I had set up and straight into a short named
folder of the root of the drive and all was well.

It's all working fine now and I'm off and running...well, baby crawling
more like it, but forwards, never the less.

Cheers,
Walter
